handlers/rootfs_image_test.go

```diff
@@ -148,3 +150,183 @@ func TestRootfsReadHeader(t *testing.T) {
 		})
 	}
 }
+
+func TestComposeHeader(t *testing.T) {
+
+	tests := map[string]struct {
+		rfs        *Rootfs
+		args       ComposeHeaderArgs
+		err        error
+		verifyFunc func(args ComposeHeaderArgs)
+	}{
+		"wrong version": {
+			rfs:  NewRootfsInstaller(),
+			args: ComposeHeaderArgs{Version: -1},
+			err:  errors.New("ComposeHeader: rootfs-version 0 not supported"),
+		},
+		"version 1 - no tar writer": {
+			rfs:  NewRootfsV1(""),
+			args: ComposeHeaderArgs{},
+			err:  errors.New("writer: tar-writer is nil"),
+		},
+		"version 2 - no tar writer": {
+			rfs:  NewRootfsV2(""),
+			args: ComposeHeaderArgs{},
+			err:  errors.New("writer: tar-writer is nil"),
+		},
+		"version 3 - no tar writer": {
+			rfs:  NewRootfsV3(""),
+			args: ComposeHeaderArgs{},
+			err:  errors.New("ComposeHeader: Payload: can not tar type-info: arch: Can not write to empty tar-writer"),
+		},
+		"version 1 - success": { // TODO - should this succeed with no update files?
```

Comment:
Not sure. Which is correct?

Reply:
I'm not sure either, but I wouldn't worry too much about it. We have completely disabled v1 support on the reading side, in anticipation of removing it altogether.